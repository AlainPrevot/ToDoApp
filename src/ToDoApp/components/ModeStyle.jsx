import "../../sass/components/ModeStyle.sass";

// eslint-disable-next-line react/prop-types
export const ModeStyle = ({ onMode }) => {
  return (
    <>
      <input
        id="mode"
        type="checkbox"
        name="modeStyle"
        className="mode"
        onClick={() => onMode()}
      />
      <label htmlFor="mode" className="modeDark"></label>
      <label htmlFor="mode" className="modeLight"></label>
    </>
  );
};
