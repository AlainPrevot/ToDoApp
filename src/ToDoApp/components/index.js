export * from './ModeStyle.jsx'
export * from './ToDoAdd.jsx'
export * from './ToDoContentListAdd.jsx'
export * from './ToDoDate.jsx'
export * from './ToDoItem'
export * from './ToDoList.jsx'
export * from './ToDoNav.jsx'
export * from './ToDoOptions.jsx'
export * from './ToDoOptionsButton.jsx'
export * from './ToDoOptionsDropdown.jsx'
