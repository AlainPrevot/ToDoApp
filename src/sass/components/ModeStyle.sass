#mode
    visibility: hidden

.modeDark, .modeLight
    background-position: center
    background-repeat: no-repeat
    background-size: cover
    cursor: pointer
    height: 36px
    position: fixed
    right: 18px
    top: 14px
    width: 36px

.modeLight
    background-image: url('../../Icons/Sol.png')
    display: none

.modeDark
    background-image: url('../../Icons/Luna.png')

#mode:checked ~ .modeLight
    display: block

#mode:checked ~ .modeDark
    display: none
