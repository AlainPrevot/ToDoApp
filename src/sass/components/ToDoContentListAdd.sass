.ToDoContentListAdd
    width: 90%
    height: 70%
    position: relative

.toDoContentListAdd__div
    height: 90%
    overflow-y: scroll

.toDoContentListAdd__div::-webkit-scrollbar
    -webkit-appearance: none

.toDoContentListAdd__div::-webkit-scrollbar:vertical
    width: 4px

.toDoContentListAdd__div::-webkit-scrollbar:horizontal
    height: 2px

.toDoContentListAdd__div::-webkit-scrollbar-thumb
    background-color: var(--Color__secondary)
    border-radius: 20px

.toDoContentListAdd__div::-webkit-scrollbar-track
    border-radius: 10px

.toDoContentListAdd__div::-webkit-scrollbar-button:increment, .toDoContentListAdd__div::-webkit-scrollbar-button
    display: none

@media only screen and (min-width: 560px)

    .toDoContentListAdd__div
        height: 80%
