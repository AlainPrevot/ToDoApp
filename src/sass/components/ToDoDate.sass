.toDoDate
    margin: 50px 0
    text-align: center

    h2
        color: white
        font-size: 2.8rem

    h6
        color: #6E9CB9
        font-size: 1.2rem
        font-weight: 300
        margin: -0.7rem
