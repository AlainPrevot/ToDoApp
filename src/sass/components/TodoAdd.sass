
.toDoAdd__input
    background-color: #151D27
    border-radius: 6px
    border: none
    color: white
    height: 46px
    margin-bottom: 20px
    outline: none
    padding-left: 38px
    width: 100%

.toDoAdd__input::placeholder
    color: #6E9CB9
