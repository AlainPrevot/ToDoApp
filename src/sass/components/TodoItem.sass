.toDoItem
    cursor: pointer
    font-size: 1.8rem
    margin-left: 8px

.toDoItem--realizado
    color: var(--Color__primary)

    .toDoItem__span
        text-decoration: line-through

.toDoItem__span
    align-items: center
    display: flex
    font-size: 1rem
    height: 100%
    position: relative
    text-transform: capitalize
    top: -3px
