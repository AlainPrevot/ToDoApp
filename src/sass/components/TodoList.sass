.toDoList
    height: 400px
    overflow-x: hidden
    overflow-y: auto

.toDoList::-webkit-scrollbar
    -webkit-appearance: none

.toDoList::-webkit-scrollbar:vertical
    width: 4px

.toDoList::-webkit-scrollbar:horizontal
    height: 2px

.toDoList::-webkit-scrollbar-thumb
    background-color: var(--Color__secondary)
    border-radius: 20px

.toDoList::-webkit-scrollbar-track
    border-radius: 10px

.toDoList::-webkit-scrollbar-button:increment, .toDoList::-webkit-scrollbar-button
    display: none
