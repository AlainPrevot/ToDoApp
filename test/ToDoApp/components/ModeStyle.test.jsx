/* eslint-disable no-undef */

import { render } from "@testing-library/react";
import { ModeStyle } from "../../../src/ToDoApp/components/ModeStyle";

describe("Pruebas en ModeStyle", () => {
  const onMode = jest.fn();
  test("El componente hace match con el Snapshot", () => {
    const { container } = render(<ModeStyle onMode={onMode} />);
    expect(container).toMatchSnapshot();
  });

  test("Al presionar los LABEL se ejecuta el evento del input", () => {
    const { container } = render(<ModeStyle onMode={onMode} />);
    const modeDark = container.getElementsByClassName("modeDark")[0];
    const modeLight = container.getElementsByClassName("modeLight")[0];
    modeDark.click();
    modeLight.click();
    expect(onMode).toHaveBeenCalledTimes(2);
  });
  test("El checkbox funciona correctamente", () => {
    const { container } = render(<ModeStyle onMode={onMode} />);
    const input = container.getElementsByClassName("mode")[0];
    expect(input.checked).toBe(false);
    input.click();
    expect(input.checked).toBe(true);
  });
});
