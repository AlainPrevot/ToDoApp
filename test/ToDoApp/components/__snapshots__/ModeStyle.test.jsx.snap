// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en ModeStyle El componente hace match con el Snapshot 1`] = `
<div>
  <input
    class="mode"
    id="mode"
    name="modeStyle"
    type="checkbox"
  />
  <label
    class="modeDark"
    for="mode"
  />
  <label
    class="modeLight"
    for="mode"
  />
</div>
`;
